transports/grpc/pipeline.go

```diff
@@ -0,0 +1,101 @@
+package grpctrans
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/raft"
+
+	transportv1 "github.com/hashicorp/raft/proto/transport/v1"
+)
+
+const (
+	// rpcMaxPipeline controls the maximum number of outstanding
+	// AppendEntries RPC calls.
+	rpcMaxPipeline = 128
```

Comment:
See #541, we probably need to at least mirror that PR in this transport if we use it to avoid the same pathology.